### DISCUSSION TITLE
Anonymous type base class

### DISCUSSION THREAD

**Discussion by Mr0N** [dotnet/csharplang#7025] [Language Ideas]
### Background and motivation

I suggest adding the ability to imitate anonymous types from classes.After compilation, so that a class of anonymous type , inherits from the class

### API Proposal

```csharp
class TestClass
{
    public Children Start()
    {
        return new
        {
            Name="282398",
            id=2
        }:Chidren;
    }

}
class Children
{

}
```

### API Usage

```csharp

```

**Comment by ghost** [comment]
I couldn't figure out the best area label to add to this issue. If you have write-permissions please help me learn by adding exactly one [area label](https://github.com/dotnet/runtime/blob/master/docs/area-owners.md).

**Comment by HaloFour** [comment]
See:

#4301 
#4148
#2517
#738

**Reply by Mr0N** [reply]
It's a shame they didn't add aliases for anonymous types for some reason) so they can be moved outside of methods

**Reply by HaloFour** [reply]
If you're going to give them an alias they're not really anonymous anymore, and you might as well define a record.
